src/components/appSwitch/AppSwitch.js

```diff
@@ -157,10 +160,17 @@ const AppSwitch = (props: LoadingProps) => {
       const identifier = goodWallet.getAccountForType('login')
 
       identifyWith(undefined, identifier)
-      showOutOfGasError()
 
-      initialize()
-      runUpdates(goodWallet, userStorage) //this needs to wait after initreg where we initialize the database
+      const isOutOfGas = await showOutOfGasError()
+
+      if (isOutOfGas) {
+        return
+      }
+
+      await initialize()
+
+      // this needs to wait after initreg where we initialize the database
+      runUpdates(goodWallet, userStorage, log)
```

Comment:
@roma-hladilka why does it breaks startup ?

@sirpy do we need to wait for updates before starting up or they could be ran 'in background' ?

Reply:
@johnsmith-gooddollar usually we dont need to block for runupdates, but there's a deadlock then it means that something is wrong and we need to fix it regardless